scipy/cluster/tests/test_vq.py

```diff
@@ -139,16 +134,16 @@ def test_py_vq(self, xp):
             xp_assert_equal(label1, xp.asarray(LABEL1, dtype=xp.int64),
                             check_dtype=False)
 
-    @skip_if_array_api
-    def test_vq(self):
+    @skip_if_array_api(np_only=True, reasons=['`_vq` only supports NumPy backend'])
+    def test_vq(self, xp):
         initc = np.concatenate([[X[0]], [X[1]], [X[2]]])
-        for tp in [np.asarray, matrix]:
+        arrays = [xp.asarray] if SCIPY_ARRAY_API else [np.asarray, matrix]
```

Comment:
not entirely sure whether we should be dropping the `matrix` check or not.

Reply:
I'll add a separate test for `matrix`.

Reply:
addressed in [cc58124](https://github.com/scipy/scipy/pull/19682/commits/cc581243827c226caebdbbd8041ad08a224b1006)